src/Heap.js

```diff
@@ -1,69 +1,83 @@
-class Heap {
-  constructor(comparator) {
-    this.arr = []
-    this.comparator = comparator
-  }
+const removeArrayItems = require('remove-array-items')
 
-  insert(item) {
-    this.arr.push(item)
-    this.callHeapify()
-  }
 
-  callHeapify() {
-    let ind = this.arr.length - 1
-    while (ind > 0) {
-      if (ind % 2 == 0) {
-        ind = (ind - 2) / 2
-      } else {
-        ind = (ind - 1) / 2
-      }
-      this.arr = heapify(this.arr, ind, this.comparator)
-    }
-  }
+// creates a new heap data structure
+function create (comparator, maxLength=1000) {
+  const arr = new Array(maxLength)
```

Comment:
Add check for validating maxLength. Eg. it shouldn't be < 1.

Reply:
good catch!

Reply:
It seems the javascript engine already throws an error, no need to handle explicitly I think:

<img width="426" alt="Screen Shot 2021-02-11 at 3 39 06 PM" src="https://user-images.githubusercontent.com/718067/107713054-50f7de00-6c7f-11eb-867a-93e9dbd5c8ad.png">

Reply:
I could add a check to ensure it's an integer though. I think that's valuable.

Reply:
resolved via 36d5f7c